lib/src/kbasesearchengine/events/handler/WorkspaceEventHandler.java

```diff
@@ -570,4 +573,108 @@ private static String toWSRefPath(final List<GUID> objectRefPath) {
+
+        // check if object is deleted and update state of latestEventType
+        command = new HashMap<>();
+        command.put("command", "listObjects");
+        command.put("params", new ListObjectsParams().withShowOnlyDeleted(wsid).withIds(Arrays.asList(wsid)));
+        final List<Tuple11> deletedObjList;
+        try {
+            deletedObjList = ws.getClient().administer(new UObject(command))
+                    .asClassInstance(List.class);
+
+            for( Tuple11 obj: deletedObjList) {
+                long deletedObjid = Long.valueOf(obj.getE1().toString()).longValue();
+                if(deletedObjid  == objid) {
+                    latestEventType = StatusEventType.DELETED;
+                    break;
+                }
+            }
+        } catch (IOException e) {
+            throw new RetriableIndexingException(e.getMessage());
+        } catch (JsonClientException e) {
+            throw new RetriableIndexingException(e.getMessage());
+        }
+
+
+        // check if name has changed and update state of lastestName
+        ObjectSpecification os = new ObjectSpecification().withRef(
+                ev.getAccessGroupId().get().toString() + "/" +
+                        ev.getAccessGroupObjectId().get());
+
+        final List<ObjectSpecification> getInfoInput = new ArrayList<>();
+        getInfoInput.add(os);
+
+        command = new HashMap<>();
+        command.put("command", "getObjectInfo");
+        command.put("params", new GetObjectInfo3Params().withObjects(getInfoInput));
+
+        final GetObjectInfo3Results objInfo;
+        try {
+            objInfo = ws.getClient().administer(new UObject(command))
+                    .asClassInstance(GetObjectInfo3Results.class);
+            String name = objInfo.getInfos().get(0).getE2();
+            if ( !Utils.isNullOrEmpty(name) ) {
+                latestName = name;
+            }
+        } catch (IOException e) {
+            throw handleException(e);
+        } catch (JsonClientException e) {
+            throw handleException(e);
+        }
+
+        // check if permissions have changed and update state of isPublic
+        try {
+            command = new HashMap<>();
+            command.put("command", "getWorkspaceInfo");
+            command.put("params", new WorkspaceIdentity()
+                    .withId((long) ev.getAccessGroupId().get()));
+
+            final String isPublic;
+
+            isPublic = ws.getClient().administer(new UObject(command))
+                    .asClassInstance(WS_INFO_TYPEREF).getE6();
+
+            if ( !Utils.isNullOrEmpty(isPublic) ) {
+                latestIsPublic = (isPublic == "n") ? true: false;
+
+            }
+        } catch (IOException | JsonClientException ex) {
+            throw new RetriableIndexingException(ex.getMessage());
+        }
+
+        StatusEvent updatedEvent = StatusEvent.
+                getBuilder(ev.getStorageObjectType().get(), ev.getTimestamp(), latestEventType).
```

Comment:
may not always have a storage object type, might just have a storage code

Reply:
If the event reached the WorkspaceEventHandler, it must have the storage object type right?

Reply:
No, for instance a DELETE_ALL_VERSIONS won't. I think the majority of the events won't.